docs/wamr: initial guide (#12477)
apache/nuttx: Documentation/applications/interpreters/wamr/index.rst

==================================
``wamr`` Webassembly Micro Runtime
==================================

> This captures how to try WAMR on NuttX.

Reviewer: This guides explains how to test WAMR on NuttX and how to get it working correctly.

Author: updated

Preparing test.wasm
===================

> Before trying WAMR, we need a test.wasm program, we need follow the "wasi-sdk" part in this `WAMR guide <https://wamr.gitbook.io/document/basics/getting-started/host_prerequsites>`_ to get the SDK and build the test.wasm program on our host.

Reviewer: Maybe you could include a simple test.wasm example here

Author: The path to WASM sample is added in the updated version.

==================================
``wamr`` Webassembly Micro Runtime
==================================

This guides explains how to try WAMR on NuttX.

Install WASI SDK
================

> We need be able to build WASM programs. this can be done by installing the ``wasi-sdk`` as shown in the `WAMR guide <https://wamr.gitbook.io/document/basics/getting-started/host_prerequsites>`_.

Reviewer 2: ```suggestion
We need to be able to build WASM programs. This can be done by installing the ``wasi-sdk`` as shown in the `WAMR guide <https://wamr.gitbook.io/document/basics/getting-started/host_prerequsites>`_.
```

Make sure the installation is reachable via ``/opt/wasi-sdk`` on your host, as this is expected by the build script of the WASM hello world sample below.

Building NuttX
==============

> We use the `sim/wamr` config to build WAMR with NuttX. Note that CMake scripts doesn't work, use makefile instead:

Reviewer 2: ```suggestion
We use the `sim/wamr` config to build WAMR with NuttX. Note that at this writing, the CMake scripts don't work for this configuration. For now, please use the makefile build instead:
```

.. code:: console

   $ cd nuttx
   $ tools/configure.sh sim/wamr
   $ make -j4
   $ ls -l nuttx
   -rwxrwxr-x 1 yf yf 4176376 Jun 10 08:11 nuttx

> This will generate a ``nutxx`` binary, also a WASM sample will be available in ``apps/interpreters/wamr/wamr/product-mini/app-samples/hello-world`` folder, we build there like below:

Reviewer 2: ```suggestion
This will generate a ``nuttx`` binary. Also a WASM sample is available in the folder ``apps/interpreters/wamr/wamr/product-mini/app-samples/hello-world``. We build the sample as follows:
```

.. code:: console

   $ cd apps/interpreters/wamr/wamr/product-mini/app-samples/hello-world
   $ ./build.sh
   $ ls -l test.wasm
   -rwxrwxr-x 1 yf yf  413 Jun 10 08:18 test.wasm

> The ``test.wasm`` program should be accessbile from NuttX later.

Reviewer 2: ```suggestion
The ``test.wasm`` program should be accessible from NuttX later.
```
